hw/ip/entropy_src/doc/_index.md

```diff
@@ -10,83 +10,143 @@ This module conforms to the [Comportable guideline for peripheral functionality.
+
+These raw noise bits are subjected to a sequence of health-checks to screen the raw signals for statistical defects which would cause any significant deviations from ideal entropy at the output of the conditioning block.
+These tests include:
+- The Repetition Count test, which screens for stuck-bits, or a complete failure of the physical noise source,
+- The Adaptive Proportion test, which screens for statistical bias in the number of 1's or 0's output by the noise source,
+- The "Bucket Test", which looks for correlations between the individual noise channels that the external noise source concatenates together to produce the raw noise sequence,
+- The "Markov Test", which looks for unexpected first-order temporal correlations between bits output by the individual noise channels,
+- The "Mailbox Test", in which raw-noise data is transferred to firmware in contiguous 1k-blocks, so that firmware can perform custom tests, and signal a failure through the same path as the other tests, and
+- Optional Vendor Specific tests, which allow silicon creators to extend the health checks by adding a top-level block external to this IP.
+
+The Repetition Count and Adaptive Proportion test are specifically recommended by SP 800-90B, and are implemented in accordance with those recommendations.
+In FIPS/CC-compliance mode, these latter three checks are performed on fixed window of data, typically consisting of 1024 bits each.
+Per the definition in SP 800-90B, the Repetition Count test does not operate on a fixed window.
+The repetition count test fails if any sequence of bits continuously asserts the same value for too many samples, as determined by the programmable threshold, regardless of whether that sequence crosses any window boundaries.
+The thresholds for these tests should be chosen to achieve a low false-positive rate (&alpha;) given a conservative estimate of the manufacturing tolerances of the physical noise source.
+The combined choice of threshold and window size then determine the false-negative rate (&beta;), or the probability of missing statistical defects at any particular magnitude.
+
+In order to compensate for the fact our tests (like *all* realistic statistical tests) have finite resolution for detecting defects, we conservatively use 1024 bits of raw digital noise to construct each 384 bit key.
+When passed through the conditioning block, the resultant entropy stream will be full entropy unless the physical noise source has encountered some statistical defect serious enough to reduce the raw min-entropy to a level below 0.375 bits of entropy per output bit.
+We choose this level as our definition of "non-tolerable statistical defects" for the purposes of evaluating this system under AIS31.
```

Comment:
Offline comment from @zi-v: "I think we should choose a much lower rate than the 384 bit strength. I believe an order of 100 bits is enough.  IMO, it is more important to keep away from false alarms".

Reply:
Hi @zi-v, 
As we mentioned in an offline meeting, we are really trying to get away from the complexity of a custom implementation of the CTR_DRBG derivation function, in lieu of having pre tested SHA conditioner.  Getting rid of the derivation function from SP 800-90A requires that we make this a full entropy source.  So we need to have 384 bits of entropy in every seed.
As far as &alpha; goes.   If we test twice, and have a single-test &alpha; of 10<sup>-12</sup> we can make for a very very low false alarm rate when we are actually producing entropy.

Reply:
I meant to say that maybe "non-tolerable statistical defects" can be lower than 384. I am not sure we need to comply with the full 384 bit under malfunction. In any case, this does not influence the design if the thresholds are configurable, so all is good.

Reply:
Hi @zi-v,
That is true about the thresholds.  However, our approach right now, was to enforce this so for NIST we could claim full-entropy and not need to use a derivation function in CTR_DRBG.   
The term here "non-tolerable" though comes from AIS31, and they do say that you have to detect this "soon" or "immediately" depending on whether you are talking about continuous or start-up tests.   And Non-tolerable for AIS would mean anything that would drop the final entropy below 0.997.

Reply:
I guess I'm confused though, when you say "I am not sure we need to comply with the full 384 bit under malfunction."  If there is a malfunction don't we always want to shut down?  Am I misunderstanding you perhaps?